cmd/inferTime.go

```diff
@@ -0,0 +1,163 @@
+	nullTime := time.Time{}
+	if tb.lastRecordedTime == nullTime {
+		errMsg := "Fatal error: gap start time is empty"
+		log.Println(errMsg)
+		err = fmt.Errorf(errMsg)
+		return err
+	}
+	if tb.nextValidTime == nullTime {
+		errMsg := "Fatal error: gap end time is empty"
+		log.Println(errMsg)
+		err = fmt.Errorf(errMsg)
+		return err
+	}
+
+	//Are the two times equal?
+	if tb.nextValidTime == tb.lastRecordedTime {
+		errMsg := "Fatal error: the start and end gap times are equal"
+		log.Println(errMsg)
+		err = fmt.Errorf(errMsg)
+		return err
+	}
+
+	//Compute the time difference
+	startTime := tb.lastRecordedTime
+	endTime := tb.nextValidTime
+	diff := endTime.Sub(startTime)
+
+	//Fail if we have a negative time difference
+	if diff.Seconds() < 0 {
+		errMsg := "Fatal error: Gap start time is later than the Gap end time"
+		log.Println(errMsg)
+		return fmt.Errorf(errMsg)
+	}
+
+	//Do we have a non null noTimeCount
+	if tb.noTimeCount < 1 {
+		errMsg := fmt.Sprintf("Fatal error: invalid number of records without time (%d)\n", tb.noTimeCount)
+		log.Println(errMsg)
+		return fmt.Errorf(errMsg)
+	}
+
+	//TODO: What should we expect as logFilePosition?
+
+	//Compute the gap
+	floatInterval := diff.Seconds() / float64(tb.noTimeCount+1)
+	tb.deltatime = int(floatInterval)
+
+	return nil
+}
+
+//storeTimeGap updates an InferTimeBLock (last valid time, nbr of records without time). It returns true if we reached the end of the time gap.
+func (tb *InferTimeBlock) storeTimeGap(logline LogLine, position int) (bool, error) {
+	var err error
+	err = nil
+
+	//ActualTime is filled if a time could be found in the FLE input
+	if logline.ActualTime != "" {
+		//Are we starting a new block
+		if tb.noTimeCount == 0 {
+			tb.lastRecordedTime = convertDateTime(logline.Date + " " + logline.ActualTime)
+			tb.logFilePosition = position
+		} else {
+			// We reached the end of the gap
+			nullTime := time.Time{}
+			if tb.lastRecordedTime == nullTime {
+				errMsg := "Fatal error: gap start time is empty"
+				log.Println(errMsg)
+				err = fmt.Errorf(errMsg)
+				return false, err
+			}
+
+			tb.nextValidTime = convertDateTime(logline.Date + " " + logline.ActualTime)
+			return true, err
+		}
+	} else {
+		//Check the data is correct.
+		nullTime := time.Time{}
+		if tb.lastRecordedTime == nullTime {
+			errMsg := "Fatal error: gap start time is empty"
+			log.Println(errMsg)
+			err = fmt.Errorf(errMsg)
+		}
+		if tb.nextValidTime != nullTime {
+			errMsg := "Fatal error: gap end time is not empty"
+			log.Println(errMsg)
+			err = fmt.Errorf(errMsg)
+		}
+
+		tb.noTimeCount++
+	}
+	return false, err
+}
+
+//convertDateTime converts the FLE date and time into a Go time structure
+func convertDateTime(dateStr string) (fullDate time.Time) {
+	const RFC3339FullDate = "2006-01-02 1504"
+
+	date, err := time.Parse(RFC3339FullDate, dateStr)
+	//error should never happen
+	if err != nil {
+		panic(err)
+	}
+
+	return date
+}
```

Comment:
```suggestion
func convertDateTime(raw string) (time.Time, error) {
	date, err := time.Parse(time.RFC3339, raw)
	if err != nil {
		return time.Time{}, fmt.Errorf("unable to parse date: %w", err)
	}

	return date, nil
}
```

So few remarks here:

- Naming the returned value is not necessarily useful. I would personally avoid it.
- The RFC3339 format is actually already declared in the `time` [package](https://golang.org/pkg/time/#pkg-constants), no need to re declare yours.
- I would return an error here, as you're parsing a time directly from your input file, which is user controlled. And one can mess up the timing 😁 
- This function actually boils down to `time.Parse` I get this right, so WDYT of removing it and use `time.Parse` directly as it does not carry any logic ?

Reply:
>Naming the returned value is not necessarily useful. I would personally avoid it.
I indeed, I am struggling with this one. One side it is ballast but on the other a well chosen name indicates the purpose of the return. But this could be an indicator of a smell in my function prototype. I will review my functions in that light: it should be self explanatory. 

> The RFC3339 format is actually already declared
Unless I missed something, I researched it and came to the conclusion that the time is not support. Time input and output as string in the system is in format "HHMM" and not as "HH:MM" as in the standard. But I will dig into the details of how the parse and format time work.

> returning error
Very good point and your proposal is very elegant. Note that I am reading in-memory data that has already been read from the file. Having a bad format here would really be unexpected but worth catching. What do you think? Foreseeing a parachute in a submarine, just in case ?

>This function actually boils down to a `time.Parse`
You are perfectly right. I don't know why I went that way. Probably thought it was going to be much more complicated when I started that part.

(Thank you very much for your very useful comments)

Reply:
> Unless I missed something, I researched it and came to the conclusion that the time is not support.

True, I was confused by the naming of the variable. Looking at the [RFC](https://www.ietf.org/rfc/rfc3339.txt) it theems that the format you're defining is not RFC3339 :thinking:  ?

> am reading in-memory data that has already been read 

Read yes, but not necessarily parsed ?

Reply:
Aaah. Very good point. It is not RFC compliant indeed and the constant name is badly chosen. Copy&Paste examples and just tweaking them a little is a very bad idea. It is not obvious and mis-leading for the casual reader.

Good comment.
The FLE input format gives (and I store it) with the date and the time separate (as there are many ways to enter that data). The conversion of these two parameters to a date/time type is because I want to handle the case where the time gap can be over a date. Date and time is entered in UTC and so US operators on the west coast have to handle the case of date change. 
So I combine the date (string) field and the time (field) in a single string to be converted to a date/time "object". Normally the separate date and time fields have been parsed and validated when filling the time gap structure. 
But on the other side, I like to be over cautious and add fuses in my code (I see myself changing the code in a couple of months/years....)

Interesting conversation.

Reply:
Simplified following the advice.